Core/GDCore/Project/ResourcesManager.h

```diff
@@ -447,6 +447,11 @@ class GD_CORE_API ResourcesManager {
    */
   std::shared_ptr<Resource> CreateResource(const gd::String& kind);
 
+  /**
+   * Get a list containing all the resources.
+   */
+  std::vector<std::shared_ptr<Resource>> GetAllResources() const { return resources; };

```

Comment:
For my previous change recommendation to work, you must also change here the return type:

```suggestion
  const std::vector<std::shared_ptr<Resource>>& GetAllResources() const { return resources; };
```

This works because you say:
- I'm returning to you a reference to a vector (so you don't have to copy it in memory).
- Because this method is `const`, the reference I give back to you is also `const` - because it's my guarantee that you (the caller) won't try to modify it.